contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -692,7 +694,7 @@ exec_stmt_exec(PLtsql_execstate *estate, PLtsql_stmt_exec *stmt)
 	if (strcmp(stmt->proc_name, "sp_describe_first_result_set") != 0)
 	{
 		if (strncmp(stmt->proc_name, "sp_", 3) == 0 && strcmp(cur_dbname, "master") != 0
-			&& (stmt->schema_name == '\0' || strncmp(stmt->schema_name, "dbo", strlen(stmt->schema_name)) == 0))
+			&& (stmt->schema_name == NULL || strncmp(stmt->schema_name, "dbo", strlen(stmt->schema_name)) == 0))
```

Comment:
stmt->schema_name == '\0'  is checking for 0 len data but will not be NULL since it may have allocated memory which is initialised as '\0'. It could also be that due to garbage value the first char is '\0' then stmt->schema_name will not be NULL
whereas stmt->schema_name == NULL means that stmt->schema_name is not pointing to any address in the memory, hence it is NULL.